charms/seldon-core/reactive/seldon_core.py

```diff
@@ -20,7 +20,9 @@ def charm_ready():
     layer.status.active('')
 
 
-@when_any('layer.docker-resource.oci-image.changed', 'config.changed')
+@when_any(
+    'layer.docker-resource.oci-image.changed', 'config.changed', 'endpoint.service-mesh.changed'
```

Comment:
You should use `endpoint.service-mesh.joined` instead, for three reasons:
1) The changed flag isn't automatically cleared, so you'll constantly have the `charm.started` flag cleared
2) It's only set once when the relation first joins, so you won't get `charm.started` cleared on every unit
3) It will be cleared when the relation goes away, so you'll get a chance to flip the flag back

Actually, on that last one, I'm not certain what `is_relation_made` will report when you get the `departed` hook for last unit as the relation is going away (which is when the `.joined` flag will be cleared).  It will know the relation is going away, but it might still report it as established until the broken hook. Maybe not a big deal, though.